include/unifex/schedule_with_subscheduler.hpp

```diff
@@ -51,10 +51,9 @@ namespace _subschedule {
       typename _detail::_return_value<remove_cvref_t<T>>::type;
 
     template <typename Scheduler>
-    using _default_result_t =
-      decltype(transform(
-        UNIFEX_DECLVAL(callable_result_t<decltype(schedule), Scheduler&>),
-        UNIFEX_DECLVAL(_return_value<Scheduler>)));
+    using _default_result_t = _tfx::sender<
+        callable_result_t<decltype(schedule), Scheduler&>,
+        _return_value<Scheduler>>;
```

Comment:
I don't think this is going to give the right result in the case that `transform()` has been customised for `schedule(Scheduler&)`.
It's assuming that `transform()` will fall-back to the default implementation.

A workaround here would be to manually call the `_tfx::sender` constructor in default implementation below, rather than calling `transform()`.